packages/gcc10.rb

```diff
@@ -1,63 +1,110 @@
+    gcc_version = version.split('-')[0]
+    # Use full gcc path to bypass ccache
+    stdout_and_stderr, status = Open3.capture2e('bash', '-c',"#{CREW_PREFIX}/bin/gcc -v 2>&1 | tail -1 | cut -d' ' -f3")
+    if status.success?
+      installed_gccver = stdout_and_stderr.chomp
+      # One gets "1:" with no gcc installed.
+      unless installed_gccver.to_s == '1:' || 
+        installed_gccver.to_s == 'No' || 
+        installed_gccver.to_s == 'not' || 
+        installed_gccver.to_s == 'gcc:' || 
+        installed_gccver.to_s == gcc_version.to_s
+        abort "GCC version #{installed_gccver} already installed.".lightgreen
+      end
+    end
   end
 
   def self.build
     # Set ccache sloppiness as per
     # https://wiki.archlinux.org/index.php/Ccache#Sloppiness
-    system "ccache --set-config=sloppiness=file_macro,locale,time_macros"
+    system 'ccache --set-config=sloppiness=file_macro,locale,time_macros'
     # Prefix ccache to path.
-    ENV['PATH'] = "#{CREW_LIB_PREFIX}/ccache/bin:#{CREW_PREFIX}/bin:/usr/bin:/bin"
+    @path = "#{CREW_LIB_PREFIX}/ccache/bin:#{CREW_PREFIX}/bin:/usr/bin:/bin"
+
+    # Install prereqs using the standard gcc method so they can be
+    # linked statically.
+    # system './contrib/download_prerequisites'
+
+    # Install newer version of gmp
+    gmp_url = "https://gmplib.org/download/gmp/gmp-#{@gmp_ver}.tar.lz"
+    gmp_sha256 = '2c7f4f0d370801b2849c48c9ef3f59553b5f1d3791d070cffb04599f9fc67b41'
+    system "curl -Ls #{gmp_url} | hashpipe sha256 #{gmp_sha256} | tar --lzip -x"
+    system "ln -sf ../gmp-#{@gmp_ver} gmp"
 
-    gcc_version = version.split("-")[0]
+    # Install newer version of isl
+    isl_url = "http://isl.gforge.inria.fr/isl-#{@isl_ver}.tar.bz2"
+    isl_sha256 = 'c58922c14ae7d0791a77932f377840890f19bc486b653fa64eba7f1026fb214d'
+    system "curl -Ls #{isl_url} | hashpipe sha256 #{isl_sha256} | tar xj"
+    system "ln -sf ../isl-#{@isl_ver} isl"
+
+    # Install newer version of mpc
+    mpc_url = "https://ftp.gnu.org/gnu/mpc/mpc-#{@mpc_ver}.tar.gz"
+    mpc_sha256 = '17503d2c395dfcf106b622dc142683c1199431d095367c6aacba6eec30340459'
+    system "curl -Ls #{mpc_url} | hashpipe sha256 #{mpc_sha256} | tar xz"
+    system "ln -sf ../mpc-#{@mpc_ver} mpc"
+
+    # Install newer version of mpfr
+    mpfr_url = "https://www.mpfr.org/mpfr-current/mpfr-#{@mpfr_ver}.tar.xz"
+    mpfr_sha256 = '0c98a3f1732ff6ca4ea690552079da9c597872d30e96ec28414ee23c95558a7f'
+    system "curl -Ls #{mpfr_url} | hashpipe sha256 #{mpfr_sha256} | tar xJ"
+    Dir.chdir "mpfr-#{@mpfr_ver}" do
+      system 'curl -Ls "https://gforge.inria.fr/scm/viewvc.php/mpfr/misc/www/mpfr-4.1.0/allpatches?revision=14491&view=co" | \
+        hashpipe sha256 dfa7d8a14ec7cb3b344cb81cfd7bd7e22aba62379941cc9110759f11172ac013 | patch -NZp1 --binary'
+    end
+    system "ln -sf ../mpfr-#{@mpfr_ver} mpfr"
```

Comment:
Why aren't we using packages `gmp`, `isl`, `mpc` and `mpfr` instead?

Reply:
I should comment this out. This was before I updated those packages.